vision_agent_tools/models/sam2.py

```diff
@@ -0,0 +1,117 @@
+import torch
+import numpy as np
+from PIL import Image
+from typing import Optional, Tuple
+from pydantic import BaseModel, Field, validate_call
+from vision_agent_tools.shared_types import BaseMLModel, Device, VideoNumpy
+from sam2.sam2_image_predictor import SAM2ImagePredictor
+
+
+class SAM2Config(BaseModel):
+    model_name: str = Field(
+        default="facebook/sam2-hiera-large",
+        description="Name of the model",
+    )
+    device: Device = Field(
+        default=Device.GPU
+        if torch.cuda.is_available()
+        else Device.MPS
+        if torch.backends.mps.is_available()
+        else Device.CPU,
+        description="Device to run the model on. Options are 'cpu', 'gpu', and 'mps'. Default is the first available GPU.",
+    )
+
+
+class SAM2Model(BaseMLModel):
```

Comment:
are we going to add a unit test as part of this PR to run this model?

Reply:
You're right! I had started an image test by testing just the basics of the inputs, do you think we should include other tests?

Updated the PR with the new file:
https://github.com/landing-ai/vision-agent-tools/blob/refactor/isolate-tool-models/tests/models/test_sam2_image.py